### PR TITLE
Add deprecation warning for ProjectReportsPluginConvention

### PR DESCRIPTION
**Pull request by donat**

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by ov7a** [comment]
@bot-gradle test this

**Comment by ov7a** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc [hunk 1]
```diff
@@ -212,6 +212,78 @@ Some of them still registers conventions for backwards compatibility.
 Registering conventions does not emit a deprecation warning yet to provide a migration window.
 Future Gradle versions will do.
 
+[[project_report_convention_deprecation]]
```
Comment (jbartok) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc:215]: Please look at what I merged in for the Java Plugin and the Application Plugin and make this part a bit more consistent with that.

**Review by jbartok — CHANGES_REQUESTED** [review]
Looks good besides my one comment.

**Review by jbartok — APPROVED** [review]

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Pull request merged**